avocado/core/runner.py

```diff
@@ -339,6 +339,10 @@ def run_test(self, test_factory, queue, summary, job_deadline=0):
         """
         proc = None
         sigtstp = multiprocessing.Lock()
+        if 'modulePath' in test_factory[1]:
+            test_path = test_factory[1]['modulePath']
+            test_module_dir = os.path.abspath(os.path.dirname(test_path))
+            sys.path.insert(0, test_module_dir)
```

Comment:
Actually I'm quite afraid this potentially could break the main Avocado process. I'd prefer if we just store the modulePath along with the pickled object in `func_at_exit` (if provided by the test) and we'd only insert it during the `func_at_exit` and remove after we executed the code.

Imagine that the test adds a path and later in code the main thread tries to import something but receives the test's module instead of the system module.